packages/pds/src/repo/prepare.ts

```diff
@@ -170,6 +170,8 @@ export const prepareCreate = async (opts: {
   }
 
   const rkey = opts.rkey || nextRkey.toString()
+  // @TODO: validate against Lexicon record 'key' type, not just overall recordkey syntax
+  ensureValidRecordKey(rkey)
```

Comment:
Since this is coarse and lexicon-unaware, I wonder if we should push this check down into the repo code.  If the PDS is lexicon-aware, it could enforce the stricter key type check as a part of record validation.

Reply:
That makes sense to me in terms of correctness.

`repo`: I'm not super confident what the blast radius of impact would be of verifying NSID+RecordKey would be in, eg, `repo/src/util.ts` `parseDataKey()`. That seems to run both on writes and reads? So could break things like the existing with-`:` record. We could initially check only in the create/write path, maybe creating a separate `parseDataKeyPermissive` as a migration/transition thing.

`PDS`: I expect PDS to be lexicon-aware enough to do generic Lexicon validation, possibly by fetching/resolving (and caching) new Lexicon files in the future. For that, passing `rkey` to the "validate record" function would make sense to me.

Reply:
@dholms any feels on this?

Reply:
I think it's a little bit more comfortable living here for the time being.

We don't do NSID validation on collection names in the repo code & that strikes me as similar to this.

So far the repo/MST is pretty unaware of these syntatic things & is primarily concerned with the tree structure & cbor encoding. Though I think there's a good argument for pushing some more lexicon/syntax awareness into it.

In short, I'd leave it here for now but consider adjusting where we draw the abstraction boundary